UefiCpuPkg/PiSmmCpuDxeSmm/X64/SmmProfileArch.c

```diff
@@ -353,15 +351,6 @@ RestorePageTableAbove4G (
   // If page entry does not existed in page table at all, create a new entry.
   //
   if (!Existed) {
-    if (IsAddressValid (PFAddress, &Nx)) {

```

Comment:
please also clean the IsAddressValid () function since it's not needed.

Reply:
it's still used in RestorePageTableAbove4G() logic, please check. I didn't change the logic of that part

Reply:
i checked the commit message:

![image](https://github.com/user-attachments/assets/8575b972-0769-4a87-a07c-3e609d86b601)

we think it must be invalid address run into the RestorePageTableAbove4G(), then below code is unnecessary. what's the reason we keep that?

![image](https://github.com/user-attachments/assets/8b2cd8ca-3e80-4f41-9176-434f0fb221a3)

Reply:
The page table that covers 4G above range may be reclaimed in ReclaimPages(). Let's keep the  IsAddressValid () and IsValidPFAddress in RestorePageTableAbove4G()

Reply:
agree, the MMIO above 4G range can be reclaimed, still need IsAddressValid () check.